pkg/server/updateResults.go

```diff
@@ -45,43 +48,87 @@ func updateResultRequest(ce *CreateExperiment) UpdateResults {
 	pm := kruize.NewProfileManager("")
 	for _, kubeobj := range ce.KubernetesObjects {
 		for _, contlist := range kubeobj.Containers {
-
-			// get queries from performanceProfile per container:
-			metricsList := pm.GetPerformanceProfileInstanceMetrics(ce.ClusterName, kubeobj.Namespace,
-				kubeobj.Name, contlist.ContainerName, ce.TrialSettings.MeasurementDuration)
-
 			//call function to parse the metrics:
 			windowSizeStr := strings.TrimSuffix(ce.TrialSettings.MeasurementDuration, "min")
 			windowSizeInt, _ := strconv.ParseInt(windowSizeStr, 10, 64)
-			startTime := time.Now()
-			endTime := startTime.Add(time.Duration(windowSizeInt) * time.Minute)
-			updateResult = UpdateResults{
-				Version:        ce.Version,
-				ExperimentName: ce.ExperimentName,
-				StartTimestamp: startTime.Format("2006-01-02T15:04:05.000Z"), //an hour ago from now
-				EndTimestamp:   endTime.Format("2006-01-02T15:04:05.000Z"),   //now
-				KubernetesObjects: []model.KubernetesObjectMetrics{
-					{
-						Type:      "deployment",
-						Name:      kubeobj.Name,
-						Namespace: kubeobj.Namespace,
-						Containers: []model.ContainerMetrics{
-							{
-								ContainerImage: contlist.ContainerImage,
-								ContainerName:  contlist.ContainerName,
-								Metrics:        metricsList,
+
+			endTimes := getTimeWindows(1, int(windowSizeInt))
+			for i := 1; i < len(endTimes); i++ {
+				startTime := endTimes[i-1]
+				endTime := endTimes[i]
+				unixTime := endTime.Unix()
+				// get queries from performanceProfile per container:
+				metricsList := pm.GetPerformanceProfileInstanceMetrics(ce.ClusterName, kubeobj.Namespace,
+					kubeobj.Name, contlist.ContainerName, ce.TrialSettings.MeasurementDuration, unixTime)
+
+				updateResult = UpdateResults{
+					Version:        ce.Version,
+					ExperimentName: ce.ExperimentName,
+					StartTimestamp: startTime.Format("2006-01-02T15:04:05.000Z"), //an hour ago from now
+					EndTimestamp:   endTime.Format("2006-01-02T15:04:05.000Z"),   //now
+					KubernetesObjects: []model.KubernetesObjectMetrics{
+						{
+							Type:      "deployment",
+							Name:      kubeobj.Name,
+							Namespace: kubeobj.Namespace,
+							Containers: []model.ContainerMetrics{
+								{
+									ContainerImage: contlist.ContainerImage,
+									ContainerName:  contlist.ContainerName,
+									Metrics:        metricsList,
+								},
 							},
 						},
 					},
-				},
+				}
+				updateResults = []UpdateResults{updateResult}
+				upJson, _ := json.Marshal(updateResults)
+				klog.V(9).Infof("Created updateResults Object %v", string(upJson))
+				err := postUpdateResult(updateResults)
+				count := 0
+				for err != nil && count < config.Cfg.RetryCount {
+					count = count + 1
+					klog.Errorf("Cannot post updateResult %s in kruize: Will retry \n", ce.ExperimentName)
+					time.Sleep(time.Duration(config.Cfg.RetryInterval) * time.Millisecond)
```

Comment:
Log the retry wait.
Consider adding backoff to slow the retries.
```suggestion
					retryWait := time.Duration(config.Cfg.RetryInterval) * time.Millisecond
					klog.Errorf("Cannot post updateResult %s in kruize: Will retry in %d ms. \n", ce.ExperimentName, retryWait)
					time.Sleep(retryWait)
```